src/pages/BuyDomains/BuyDomains.tsx

```diff
@@ -0,0 +1,89 @@
+import { FC, useMemo } from 'react';
+import classNames from 'classnames/bind';
+import TextInput from 'zero-ui/src/components/Input/TextInput';
+import Button from 'zero-ui/src/components/Button';
+import { AppProps } from '../../types';
+import { IconZero } from '../../components';
+import { DEFAULT_NETWORK_PROTOCAL } from '../../constants/network';
+import { MIN_DOMAIN_NAME_LENGTH } from '../../constants/domains';
+import { useDomainSearch } from '../../hooks/useDomainSearch';
+import styles from './BuyDomains.module.scss';
+
+const cx = classNames.bind(styles);
+
+export const BuyDomains: FC<AppProps> = ({ provider, route }) => {
+	const domainSearch = useDomainSearch();
+
+	const handleOnChange = (value: string) => {
+		let correctValue = value;
+		if (DEFAULT_NETWORK_PROTOCAL.startsWith(value)) {
+			correctValue = '';
+		}
+		correctValue = correctValue
+			.replace(DEFAULT_NETWORK_PROTOCAL, '')
+			.replace(/[^a-zA-Z0-9]/g, '')
+			.replace(/\s+/g, '');
+		domainSearch.setPattern(correctValue);
+	};
+
+	const status = useMemo(() => {
+		const isReady =
+			domainSearch.pattern.length >= MIN_DOMAIN_NAME_LENGTH &&
+			!domainSearch.isLoading;
+		const isValid = isReady && !domainSearch.exactMatch;
+		const isInvalid = isReady && Boolean(domainSearch.exactMatch);
+
+		return {
+			isReady,
+			isValid,
+			isInvalid,
+		};
+	}, [domainSearch]);
```

Comment:
These values don't need to be memoized, as they're simple calculations. You don't need to memoize unless you're trying to avoid an expensive/long-running computation.